### DISCUSSION TITLE
Does kwok support get metrics similar to node-exporter to determine scheduling?

### DISCUSSION THREAD

**Discussion by ShyunnY** [kubernetes-sigs/kwok#1222] [Q&A]
I am currently developing a custom scheduler, let's call it `LoadNode` plugin!

This plugin mainly uses `prometheus/client` to get the CPU load of each Node stored in `prometheus` in real time, and then performs a series of algorithms to finally calculate the score of the Pod in the current Node.

Before this, I found that it was a painful thing to test. We can only test in a real test cluster, and cannot simulate through some localized cluster components (such as: kind, minikube). By chance, I learned about the kwok component.

I urgently want to know whether kwok can meet my above testing needs? Thank you very much!

**Comment by wzshiming** [comment]
Currently, node-exporter is not supported, only metrics-server is supported, maybe you can consider getting CPU load from metrics-server.

https://kwok.sigs.k8s.io/docs/technical-outcomes/scalability/scale-using-hpa/

**Reply by ShyunnY** [reply]
I may not understand the documentation. What I want is:
+ `kwok` can use the external scheduler when starting
+ and the external scheduler can get fake `kwok` metrics through `rest`

It seems that these two points cannot be achieved now?

**Reply by wzshiming** [reply]
I think [kube-scheduler-simulator](https://github.com/kubernetes-sigs/kube-scheduler-simulator)/[e2e-framework](https://github.com/kubernetes-sigs/e2e-framework/blob/main/examples/kwok/kwok_with_config/main_test.go) has a example that can do what you want.
